libraries/DataFlash/LogFile.cpp

```diff
@@ -1437,20 +1437,23 @@ void DataFlash_Class::Log_Write_Current_instance(const uint64_t time_us,
 // Write an Current data packet
 void DataFlash_Class::Log_Write_Current()
 {
+    // Big painful assert to ensure that logging won't produce suprising results when the
+    // number of battery monitors changes, does have the built in expectation that
+    // LOG_COMPASS_MSG follows the last LOG_CURRENT_CELLSx_MSG
+    static_assert(((LOG_CURRENT_MSG + AP_BATT_MONITOR_MAX_INSTANCES) == LOG_CURRENT_CELLS_MSG) &&
+                  ((LOG_CURRENT_CELLS_MSG + AP_BATT_MONITOR_MAX_INSTANCES) == LOG_COMPASS_MSG),
```

Comment:
Yeah, that was my position as well. It's ugly, and I apologize for it, but it beats ignoring the situation entirely. If we do instances this can all go away :)

Reply:
Works for me.